components/EventPage/JoinEvent.tsx

```diff
@@ -0,0 +1,105 @@
+import Box from '@mui/material/Box'
+import Button from '@mui/material/Button'
+import CalendarTodayIcon from '@mui/icons-material/CalendarToday'
+import AccessTimeIcon from '@mui/icons-material/AccessTime'
+import EventAvailableIcon from '@mui/icons-material/EventAvailable'
+import Typography from '@mui/material/Typography'
+import { styled } from '@mui/material/styles'
+import Stack from '@mui/material/Stack'
+import { grey } from '@mui/material/colors'
+import TwitterIcon from '@mui/icons-material/Twitter'
+import LinkedInIcon from '@mui/icons-material/LinkedIn'
+import Tooltip, { tooltipClasses, TooltipProps } from '@mui/material/Tooltip'
+import React from 'react'
+
+const CustomTooltip = styled(({ className, ...props }: TooltipProps) => (
+  <Tooltip {...props} arrow classes={{ popper: className }} />
+))(({ theme }) => ({
+  [`& .${tooltipClasses.tooltip}`]: {
+    backgroundColor: '#fff',
+    color: 'rgba(0, 0, 0, 0.87)',
+    maxWidth: 220,
+    fontSize: theme.typography.pxToRem(12),
+    border: '2px solid #dadde9',
+  },
+}))
+
+const CustomLink = styled('a')(({ theme }) => ({
```

Comment:
Didn't remove this component as we are using this same styling in multiple positions.
Copy-pasting the same styles again and again to make the codebase larger isn't the ideal solution, if we can refactor the code with some components, then I think we should proceed on that way only.

Though I have removed a few components and added the styles directly using props, those styles weren't repeating, but here the case is different.

Reply:
Maybe create a HOC or come up with some solution.

Reply:
Yes, sure, extracted the component and the logic